shared/src/main/scala/amf/core/internal/errorhandling/JsonErrorHandler.scala

```diff
@@ -1,12 +1,15 @@
 package amf.core.internal.errorhandling
 
 import amf.core.client.scala.errorhandling.AMFErrorHandler
+import amf.core.internal.plugins.syntax.SYamlAMFParserErrorHandler
 import amf.core.internal.validation.CoreValidations.SyamlWarning
 import org.mulesoft.lexer.SourceLocation
-import org.yaml.model.{DefaultJsonErrorHandler, SyamlException}
+import org.yaml.model.{DefaultJsonErrorHandler, ParseErrorHandler, SyamlException}
 
-case class JsonErrorHandler(override val errorHandler: AMFErrorHandler) extends DefaultJsonErrorHandler {
+case class JsonErrorHandler(eh: AMFErrorHandler) extends DefaultJsonErrorHandler {
```

Comment:
Great change, adapter was the way to go. Would probably change the name though to signal that this is Syaml specific. Probably `SyamlJsonErrorHandler` or `SyamlJsonErrorHandlerAdapter`. Doesn't move the needle though